plugins/cinema4d/import_cast.pyp

```diff
@@ -402,7 +402,7 @@ def importSkeletonNode(modelNull, skeleton):
         newBone.SetAbsScale(scale)
 
         handles[i] = newBone
-        boneIndexes[bone.Hash()] = newBone
+        boneIndexes[bone.Hash() or i] = newBone
```

Comment:
There _has_ to be a bone hash for every bone.

Reply:
Hash will not work on Apex Legends models and for how the IK is made right now it can't use index.
I'd be just too scared of using Name(), cause of the small risk of having two bones with the same name. 

Do you happen to have a better idea than those?

```
# Apex Legends bone.Hash() always returns 0
boneIndexes[bone.Hash() if bone.Hash() != 0 else i] = newBone
```
or returning two values, which looks really bad in my opinion
```
boneIndexes[i] = newBone
boneHashes[boneHash()] = newBone
```

Reply:
Just read your comment again "which may not be unique if the bone isn't referenced outside of weights" 
Maybe returning both would be the way, but is there a more "professional" way of doing it?
Else I try to think of another solution.

Reply:
Guess doing a proper check if something is unique or not?

Reply:
Apex Legends bone.Hash() always equals 0, so I came up with some ideas.

First to just check if it's 0 or not, but I don't really trust this solution.
```py
for i, bone in enumerate(bones):
    boneHash = bone.Hash()
    boneIndexes[boneHash if boneHash !=0 else i] = newBone
```
or

Second see if a boneHash already exists, but you said "may not be unique". I'm not sure if it counts as "It can't be both".
```py
seenHashes = set()
for i, bone in enumerate(bones):
    boneHash = bone.Hash()
    boneIndexes[boneHash if boneHash not in seenHashes else i] = newBone
    seenHashes.add(boneHash)
```

Another idea would be a check if  the hash is used outside of the weight or using names.
Not sure how efficient and trustworthy those are.

Please let me know what you have in mind!


Deleted previous comments to reduce spam.

Reply:
Decided to use Names() for now, since the Blender addon also makes use of that.